lib/modules.nix

```diff
@@ -186,8 +187,12 @@ rec {
 
       # Like unifyModuleSyntax, but also imports paths and calls functions if necessary
       loadModule = args: fallbackFile: fallbackKey: m:
+        let defs = [{ file = fallbackFile; value = m; }];
+        in
         if isFunction m || isAttrs m then
           unifyModuleSyntax fallbackFile fallbackKey (applyIfFunction fallbackKey m args)
+        else if isList m then
+          throw "Module imports can't be nested lists. Perhaps you meant to remove one level of lists? Definitions: ${showDefs defs}"
```

Comment:
The outer `let in` actually has a cost to Nix, even if `defs` isn't evaluated. So to improve that you should move the definition of `defs` to the `isList` case instead.

Reply:
Thanks, I wasn't actually sure whether the cost would be significant for this path.